azurerm/resource_arm_iothub_shared_access_policy.go

```diff
@@ -0,0 +1,356 @@
+	}
+
+	d.SetId(resourceId)
+
+	return resourceArmIotHubSharedAccessPolicyRead(d, meta)
+}
+
+type accessRights struct {
+	registryRead   bool
+	registryWrite  bool
+	serviceConnect bool
+	deviceConnect  bool
+}
+
+func expandAccessRights(d *schema.ResourceData) string {
+
+	var possibleAccessRights = []struct {
+		schema string
+		right  string
+	}{
+		{"registry_read", "RegistryRead"},
+		{"registry_write", "RegistryWrite"},
+		{"service_connect", "ServiceConnect"},
+		{"device_connect", "DeviceConnect"},
+	}
+	actualRights := make([]string, 0)
+	// iteration order is important here, so we cannot use a map
+	for _, possibleRight := range possibleAccessRights {
+		if d.Get(possibleRight.schema).(bool) {
+			actualRights = append(actualRights, possibleRight.right)
+		}
+	}
+	strRights := strings.Join(actualRights, ", ")
+	return strRights
+}
+
+func flattenAccessRights(r devices.AccessRights) accessRights {
+	rights := accessRights{
+		registryRead:   false,
+		registryWrite:  false,
+		deviceConnect:  false,
+		serviceConnect: false,
+	}
+
+	actualAccessRights := strings.Split(string(r), ",")
+
+	for _, right := range actualAccessRights {
+		switch strings.ToLower(strings.Trim(right, " ")) {
+		case "registrywrite":
+			rights.registryWrite = true
+			// RegistryWrite implies RegistryRead.
+			// What's more, creating a Access Policy with both RegistryRead and RegistryWrite
+			// only really sets RegistryWrite permission, which then also implies RedistryRead
+			fallthrough
+		case "registryread":
+			rights.registryRead = true
+		case "deviceconnect":
+			rights.deviceConnect = true
+		case "serviceconnect":
+			rights.serviceConnect = true
+		}
+	}
+
+	return rights
+}
+
+func resourceArmIotHubSharedAccessPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).iothubResourceClient
+	ctx := meta.(*ArmClient).StopContext
+
+	parsedIothubSAPId, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resourceGroup := parsedIothubSAPId.ResourceGroup
+	iothubName := parsedIothubSAPId.Path["IotHubs"]
+	keyName := parsedIothubSAPId.Path["IotHubKeys"]
+
+	accessPolicy, err := client.GetKeysForKeyName(ctx, resourceGroup, iothubName, keyName)
+	if err != nil {
+		if utils.ResponseWasNotFound(accessPolicy.Response) {
+			log.Printf("[DEBUG] Shared Access Policy %q was not found on IotHub %q (Resource Group %q) - removing from state", keyName, iothubName, resourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error loading IotHub Shared Access Policy %q (Resource Group %q): %+v", iothubName, resourceGroup, err)
+	}
+
+	iothub, err := client.Get(ctx, resourceGroup, iothubName)
+	if err != nil {
+		return fmt.Errorf("Error loading IotHub %q (Resource Group %q): %+v", iothubName, resourceGroup, err)
+	}
+
+	_ = d.Set("name", keyName)
```

Comment:
I don't think we need the _ =?  🤔 
```suggestion
d.Set("name", keyName)
```

Reply:
fixed